src/fft/fft.jl

```diff
@@ -0,0 +1,388 @@
+import AbstractFFTs: plan_fft, plan_fft!, plan_bfft, plan_bfft!,
+    plan_rfft, plan_brfft, plan_inv, normalization, fft, bfft, ifft, rfft,
+    Plan, ScaledPlan
+
+using LinearAlgebra
+
+const ROCFFT_FORWARD = true
+const ROCFFT_INVERSE = false
+
+abstract type ROCFFTPlan{T,K,inplace} <: Plan{T} end
+
+Base.unsafe_convert(::Type{rocfft_plan}, p::ROCFFTPlan) = p.handle
+
+function unsafe_free!(plan::ROCFFTPlan)
+    rocfft_plan_destroy(plan.handle)
+    rocfft_execution_info_destroy(plan.execution_info)
+    # hsa_memory_free(plan.workarea.handle) |> check  # needed?
```

Comment:
Forgot to remove it (HSA was complaining). I added it initially, since over at CuArrays they are freeing the workspace, but they have some custom memory management.